server/src/graphql/mutations/createAppGithub.ts

```diff
@@ -34,37 +40,38 @@ export const createAppGithub: MutationResolvers['createAppGithub'] = async (
 
   const octokit = new Octokit({});
 
-  let repo;
-
   const repoData = getRepoData(input.gitRepoUrl);
-  try {
-    repo = await octokit.repos.get({
-      owner: repoData.owner,
-      repo: repoData.repoName,
-    });
-  } catch (error) {
-    console.log(error);
+
+  const repo = await octokit.repos.get({
+    owner: repoData.owner,
+    repo: repoData.repoName,
+  });
+
+  const branch = await octokit.repos.getBranch({
+    owner: repoData.owner,
+    repo: repoData.repoName,
+    branch: input.branchName ? input.branchName : 'main',
+  });
+
+  if (!branch.url) {
+    throw new Error("There's no such branch for this repository");
   }
 
   const ssh = await sshConnect();
 
   const dokkuApp = await dokku.apps.create(ssh, input.name);
 
-  const hash = crypto
-    .createHash('sha256')
-    .update(`${input.name}` + `${userId}`, 'utf8')
-    .digest('hex')
-    .slice(0, 40);
+  const randomToken = generateRandomToken(40);
 
   const app = await prisma.app.create({
     data: {
       name: input.name,
       type: 'GITHUB',
       AppMetaGithub: {
         create: {
-          repoUrl: input.gitRepoUrl,
+          repoUrl: `${repoData.owner}/${repoData.repoName}`,
```

Comment:
do we want to change the name of this field?
If we want to apply the github naming we can have 2 fields, one "owner" and one "name"

Reply:
Yeah, I guess it makes sense to have two fields. Will create a migration in couple of min.